ci(release): Sync winget-pkgs fork before publishing

### Description 

Take a look at - https://github.com/coder/coder/actions/runs/7611646446/job/20727993997 (and other release runs)

Publishing to winget failed because:

```
Submitting pull request for manifest...

Unable to create a reference to the forked repository. This can be caused when the forked repository is behind by too many commits. Sync your fork and try again.
Error: Process completed with exit code 1.
```
### Steps to fix

This will reproduce for subsequent releases/runs until the fork used for submitting PR is kept in sync with upstream.

i.e., https://github.com/cdrci/winget-pkgs in this case

WingetCreate does attempt to sync the fork before making a PR, but that can fail if the fork is behind too many commits.

Suggest to make it as part of CI to sync the fork (through `gh` CLI?) before winget publish step to avoid this in the future

### Upstream issue

Related to:
- https://github.com/microsoft/winget-create/issues/502

--- a/.github/workflows/release.yaml
+++ b/.github/workflows/release.yaml
@@ -408,6 +408,9 @@ jobs:
     if: ${{ !inputs.dry_run }}
 
     steps:
+      - name: Sync fork
+        run: gh repo sync cdrci/winget-pkgs -b master
+        
       - name: Checkout
         uses: actions/checkout@v4
         with:

--- a/.github/workflows/release.yaml
+++ b/.github/workflows/release.yaml
@@ -412,7 +412,7 @@ jobs:
         run: gh repo sync cdrci/winget-pkgs -b master
         env:
           GH_TOKEN: ${{ secrets.WINGET_GH_TOKEN }}
-        
+
       - name: Checkout
         uses: actions/checkout@v4
         with:

--- a/.github/workflows/release.yaml
+++ b/.github/workflows/release.yaml
@@ -410,6 +410,8 @@ jobs:
     steps:
       - name: Sync fork
         run: gh repo sync cdrci/winget-pkgs -b master
+        env:
+          GH_TOKEN: ${{ secrets.WINGET_GH_TOKEN }}
         
       - name: Checkout
         uses: actions/checkout@v4

